keras/engine/topology.py

```diff
@@ -1845,12 +1845,12 @@ def get_layer(self, name=None, index=None):
         else:
             if not name:
                 raise ValueError('Provide either a layer name or layer index.')
-        layer = None
+
         for layer in self.layers:
             if layer.name == name:
                 return layer
-        if not layer:
-            raise ValueError('No such layer: ' + name)
+
+        raise ValueError('No such layer: ' + name)
```

Comment:
This is an improvement but using the "else" clause on the for statement might be even clearer (http://book.pythontips.com/en/latest/for_-_else.html)

for layer in self.layers:
   if layer.name == name:
       return layer
else:
  raise ValueError("No such layer: " + name)